File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        
        res,i,j = 0,0,len(height)-1
        while i<j:
            res = max(res,(j-i)*(min(height[j],height[i])))
            if height[i]<=height[j]:
                i+=1
            else:
                j-=1
        return res
            
         
        

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        
        int res = 0, j = height.size()-1, i = 0;
        while (i<j){
            res = max(res,(min(height[i],height[j]))*(j-i));
            
            if (height[i]>height[j]){
                j--;
            }else{
                i++;
            }
        }
        return res;
    }
};